theano/tensor/nnet/tests/test_abstract_conv.py

```diff
@@ -24,82 +23,99 @@
 from theano.tensor.nnet.ConvTransp3D import ConvTransp3D
 
 
-def conv_corr(inputs, filters, border_mode="valid", subsample=(1, 1),
-              conv_mode='conv'):
+def conv_corr(inputs, filters, border_mode="valid",
+              subsample=(1, 1), conv_mode='conv',
+              filter_dilation=(1, 1)):
     if conv_mode == 'conv':
         filters = filters[:, :, ::-1, ::-1]
-    return corr.CorrMM(border_mode, subsample)(inputs, filters)
-
-
-def conv_corr_gw(inputs, topgrad, filters_shape, border_mode="valid",
-                 subsample=(1, 1), conv_mode='conv'):
-    rval = corr.CorrMM_gradWeights(border_mode, subsample)(inputs, topgrad,
-                                                           filters_shape[2:])
+    return corr.CorrMM(border_mode,
+                       subsample,
+                       filter_dilation)(inputs, filters)
+
+
+def conv_corr_gw(inputs, topgrad, filters_shape,
+                 border_mode="valid", subsample=(1, 1),
+                 conv_mode='conv', filter_dilation=(1, 1)):
+    rval = corr.CorrMM_gradWeights(border_mode,
+                                   subsample,
+                                   filter_dilation)(inputs, topgrad,
+                                                    filters_shape[2:])
     if conv_mode == 'conv':
         rval = rval[:, :, ::-1, ::-1]
     return rval
 
 
-def conv_corr_gi(filters, topgrad, inputs_shape, border_mode="valid",
-                 subsample=(1, 1), conv_mode='conv'):
+def conv_corr_gi(filters, topgrad, inputs_shape,
+                 border_mode="valid", subsample=(1, 1),
+                 conv_mode='conv', filter_dilation=(1, 1)):
     if conv_mode == 'conv':
         filters = filters[:, :, ::-1, ::-1]
-    return corr.CorrMM_gradInputs(border_mode, subsample)(filters, topgrad,
-                                                          inputs_shape[2:])
+    return corr.CorrMM_gradInputs(border_mode,
+                                  subsample,
+                                  filter_dilation)(filters,
+                                                   topgrad,
+                                                   inputs_shape[2:])
 
 
 class TestGetConvOutShape(unittest.TestCase):
     def test_basic(self):
-        image_shape, kernel_shape = (3, 2, 8, 9), (4, 2, 5, 6)
+        image_shape, kernel_shape = (3, 2, 12, 9), (4, 2, 5, 6)
```

Comment:
Oh! There was a merge conflict and i didn't resolve it correctly before. I'm sorry about this.